### DISCUSSION TITLE
Ingredient Metadata

### DISCUSSION THREAD

**Discussion by dfrankland** [cooklang/spec#53] [Ideas]
The shopping list spec is a sort of metadata for ingredients.

Is there an example usage of both a shopping list and recipe?

It'd be nice to be able to tag ingredients by their their grocery section, places / websites to buy them, their allergy information, etc. Perhaps a more robust spec around this would be beneficial.

**Comment by umgefahren** [comment]
If I've understood it right, you don't directly declare the ingredients nor do you directly write the shopping list. The genius is the fact you don't have to do that. The CLI generates the shopping list directly from the recipes and the ingredients are directly inferred from the recipe.

There is no need for metadata definition of ingredients.

**Comment by dfrankland** [comment]
I suppose calling it a shopping list is a bit of a misnomer and is not the primary focus of this issue.

The shopping list spec specifies groups of ingredients and gives the groups names. The ingredient list can be created from just the ingredients specified within the recipe, but the other information related to those ingredients is still unspecified.

**Reply by gvlx** [reply]
Linking to #129

**Comment by LSWdigital** [comment]
At the moment it seems to be a weak type system.

Personally I think using set theory would be better than type theory.
